packages/core/src/pagination/pagination-button.element.ts

```diff
@@ -0,0 +1,121 @@
+ *   <span aria-label="current page">1 / 3</span>
+ *   <cds-pagination-button action="next"></cds-pagination-button>
+ *   <cds-pagination-button action="last"></cds-pagination-button>
+ * </cds-pagination>
+ * ```
+ *
+ * @element cds-pagination-button
+ * @slot
+ * @slot cds-icon
+ * @cssprop --background
+ * @cssprop --padding
+ * @cssprop --border-color
+ * @cssprop --border-width
+ * @cssprop --border-radius
+ * @cssprop --box-shadow
+ * @cssprop --disabled-background
+ * @cssprop --disabled-color
+ * @cssprop --min-height
+ * @cssprop --min-width
+ */
+
+export class CdsPaginationButton extends CdsBaseButton {
+  /**
+   * @type {first | prev | next | last}
+   * Sets the action from a predefined list of actions
+   */
+  @property({ type: String, reflect: true }) action: CdsPaginationButtonAction;
+
+  @property({ type: String, reflect: true }) ariaLabel: string;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.type = 'button';
+  }
+
+  @querySlot('cds-icon', { assign: 'cds-icon-slot' }) private cdsIcon: HTMLElement;
+
+  private get customContent() {
+    return !this.action && !this.cdsIcon;
+  }
+
+  render() {
+    return html`
+      <div class="private-host" cds-layout="horizontal align:center ${this.customContent ? 'p-x:sm' : ''}">
+        <slot name="cds-icon-slot">
+          ${this.action === CdsPaginationButtonAction.Next
+            ? html`<cds-icon shape="angle" direction="right"></cds-icon>`
+            : ''}
+          ${this.action === CdsPaginationButtonAction.Last
+            ? html`<cds-icon shape="step-forward-2" direction="up"></cds-icon>`
+            : ''}
+          ${this.action === CdsPaginationButtonAction.Previous
+            ? html`<cds-icon shape="angle" direction="left"></cds-icon>`
+            : ''}
+          ${this.action === CdsPaginationButtonAction.First
+            ? html`<cds-icon shape="step-forward-2" direction="down"></cds-icon>`
+            : ''}
+        </slot>
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  static get styles() {
+    return [baseStyles, styles];
+  }
+
+  updated(props: Map<string, any>) {
+    super.updated(props);
+    this.setupAriaLabel(props);
+  }
+
+  private setupAriaLabel(props: Map<string, any>) {
+    if (!props.has('ariaLabel')) {
+      switch (this.action) {
+        case CdsPaginationButtonAction.First:
+          this.ariaLabel = I18nService.keys.pagination.firstButtonAriaLabel;
+          break;
+        case CdsPaginationButtonAction.Previous:
+          this.ariaLabel = I18nService.keys.pagination.prevButtonAriaLabel;
+          break;
+        case CdsPaginationButtonAction.Next:
+          this.ariaLabel = I18nService.keys.pagination.nextButtonAriaLabel;
+          break;
+        case CdsPaginationButtonAction.Last:
+          this.ariaLabel = I18nService.keys.pagination.lastButtonAriaLabel;
+          break;
+        default:
+        // do nothing
+      }
+    }
+  }
```

Comment:
I think this logic may have an unintented bug.

Let's say the component's property such as `disabled` (inherited from `CdsBaseButton`) is changed sometime in its lifecycle. I believe this will trigger the aria-label to be re-assigned to a pre-assigned value from our service and would override a previously set custom aria-label if the user supplied one through the attribute (i.e. `<cds-pagination-button aria-label='my-custom-label'>...</cds-pagination-button>`). Of course this isn't a problem if they used the `cds-i18n` attribute to set the custom label but I think this goes back to the question about whether to use the service in this component. I'd say no since this is a property directly on the host and not inside the template as mentioned in my other comment in this PR).

Reply:
Since we've configured the property to reflect the aria-label attribute, that should be fine, right?

```
@property({ type: String, reflect: true }) ariaLabel
```

I have also set a custom `aria-label` value on the button and tested whether that user-provided text is being overridden or not:

https://github.com/vmware/clarity/pull/5706/files#diff-9299cc10edb451e7e097efac8e6b7ea84c1d95dd3a5ad996f3f98399028bb0aaR116-R125

Reply:
I think @jeeyun is correct about an unintended bug. It goes something like this...

```
Everytime the lit element update cycle runs and "ariaLabel" is not present in the Map of updated properties this switch statement will reassign "ariaLabel" even if it was already assigned a value.
```

But the more important matter is that I don't think we need this. We don't need any default set here. We only need for "ariaLabel" to be a required property so that people are warned if it is not present.

Reply:
Ooh I see now. But yes, we no longer need default ariaLabel values.